x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/index.tsx

```diff
@@ -274,14 +274,16 @@ export const CreatePackagePolicySinglePage: CreatePackagePolicyParams = ({
         <Loading />
       ) : packageInfo ? (
         <>
-          <StepDefinePackagePolicy
-            agentPolicy={agentPolicy}
-            packageInfo={packageInfo}
-            packagePolicy={packagePolicy}
-            updatePackagePolicy={updatePackagePolicy}
-            validationResults={validationResults!}
-            submitAttempted={formState === 'INVALID'}
-          />
+          {packageInfo.name !== 'cloud_security_posture' && (
+            <StepDefinePackagePolicy
+              agentPolicy={agentPolicy}
+              packageInfo={packageInfo}
+              packagePolicy={packagePolicy}
+              updatePackagePolicy={updatePackagePolicy}
+              validationResults={validationResults!}
+              submitAttempted={formState === 'INVALID'}
+            />
+          )}
```

Comment:
1. we need to not render this step so we can customize it through existing UI extension points (as per our [design](https://www.figma.com/file/QITLtzCQWAJCOW2xNsnO81/CSP-Visualizations?node-id=1080%3A345361&t=w4FA8SPlna0QfXDR-4))
2. this hard-coded condition can be replaced by checking for a new property (maybe `hideDefinePackagePolicy` ?) on `PackagePolicyCreateExtension` and `PackagePolicyEditExtension`
3. went with the most simple solution, deferring to code owners

Reply:
I think we should create a new extension point and UI extension type(s) like you've mentioned. Since we already have a CSP Kibana plugin, we should be able to register a extension in that plugin's start lifecycle (see other usages of Fleet UI extensions for example). 

We'll need to make sure that when a corresponding extension is present (I'm fine with adding a `hideDefinePackagePolicy` option to the extension contract), the form renders as expected and that we don't break the form for other non-extension providing integrations. 

Let me know if there's anything we can help with in terms of getting that UI extension created.

Reply:
It seems weird to me to hide the whole block, there is still some fields like `namespace` that could make sense to be customizable for the user no? 

It's not possible to change the design to still have the two step, but configure a UI extension to remove the global variables from `StepDefinePackagePolicy`. This way name, description, namespace are still managed by Fleet UI

Reply:
@nchaulet - i'm assuming you meant changing the design a bit? AFAIK it's intentional that we customize the `name`/`description` to match the rest of our form, and since we don't use the other options we'd prefer to hide them. (cc @tinnytintin10)



@kpollich  - i've opened a draft [PR](https://github.com/elastic/kibana/pull/149653) with some questions on creating the new extension point. would appreciate an early feedback. if possible, i'd like to get this merged before FF. in the mean time, wdyt about us proceeding with the hard coded solution for now?

Reply:
@nchaulet, on top of the reasons @orouz mentioned, the `namespace` field, in particular, is overloaded and causes confusion for our users who want to deploy the KSPM integration. Most users assume it has something to do with their namespace in their k8s environments [when it does not](https://www.elastic.co/guide/en/fleet/master/data-streams.html)

This, coupled with a lack of requirements that necessitated these and the other fields under advanced, I want to hide these fields until there is an explicit reason to have them appear; otherwise, we're just providing users with ways to shoot themselves in the foot.

Reply:
One more note on namespaces, with the way "latest" findings are generated (by a transform that runs on the default datastream sent by cloudbeat), enabling users to modify the `namespace` value (which changes the name of the data stream) could lead to a transform running on an index/datastream that doesn't exist; if this were true, it leads to a poor UX where the integration could work and post data back, but the users see nothing in our product. 

cc @orouz @kfirpeled @oren-zohar